From 622d2d3b4e2ffb3f5e24e6b14b72f1b446d9ec53 Mon Sep 17 00:00:00 2001
From: Stephane Viau <sviau@codeaurora.org>
Date: Wed, 29 Apr 2015 15:57:29 -0400
Subject: [PATCH 10/44] drm/msm: Switch to qcom KGSL driver

Switch to qcom KGSL driver in drm msm driver when it's available.

Change-Id: I031eeec6378b6ce436c09c1a8388401e7ff48f52
Signed-off-by: Stephane Viau <sviau@codeaurora.org>
---
 drivers/gpu/Makefile          |  3 ++-
 drivers/gpu/drm/msm/Makefile  | 12 ++++++++----
 drivers/gpu/drm/msm/msm_drv.c | 10 ++++++++++
 3 files changed, 20 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/Makefile b/drivers/gpu/Makefile
index 1f26b1d..192fa99 100644
--- a/drivers/gpu/Makefile
+++ b/drivers/gpu/Makefile
@@ -1,3 +1,4 @@
-obj-y			+= drm/ vga/ msm/
+obj-y			+= drm/ vga/
 obj-$(CONFIG_TEGRA_HOST1X)	+= host1x/
 obj-$(CONFIG_IMX_IPUV3_CORE)	+= ipu-v3/
+obj-$(CONFIG_MSM_KGSL)		+= msm/
diff --git a/drivers/gpu/drm/msm/Makefile b/drivers/gpu/drm/msm/Makefile
index 14d167e..5f3bd50 100644
--- a/drivers/gpu/drm/msm/Makefile
+++ b/drivers/gpu/drm/msm/Makefile
@@ -2,10 +2,6 @@ ccflags-y := -Iinclude/drm -Idrivers/gpu/drm/msm
 ccflags-$(CONFIG_DRM_MSM_DSI_PLL) += -Idrivers/gpu/drm/msm/dsi
 
 msm-y := \
-	adreno/adreno_device.o \
-	adreno/adreno_gpu.o \
-	adreno/a3xx_gpu.o \
-	adreno/a4xx_gpu.o \
 	hdmi/hdmi.o \
 	hdmi/hdmi_audio.o \
 	hdmi/hdmi_bridge.o \
@@ -50,6 +46,14 @@ msm-y := \
 	msm_rd.o \
 	msm_ringbuffer.o
 
+# use drm gpu driver only if qcom_kgsl driver not available
+ifneq ($(CONFIG_MSM_KGSL),y)
+msm-y += adreno/adreno_device.o \
+	adreno/adreno_gpu.o \
+	adreno/a3xx_gpu.o \
+	adreno/a4xx_gpu.o
+endif
+
 msm-$(CONFIG_DRM_MSM_FBDEV) += msm_fbdev.o
 msm-$(CONFIG_COMMON_CLK) += mdp/mdp4/mdp4_lvds_pll.o
 
diff --git a/drivers/gpu/drm/msm/msm_drv.c b/drivers/gpu/drm/msm/msm_drv.c
index 9d57dbf..4df58b4 100644
--- a/drivers/gpu/drm/msm/msm_drv.c
+++ b/drivers/gpu/drm/msm/msm_drv.c
@@ -441,6 +441,7 @@ fail:
 	return ret;
 }
 
+#ifndef CONFIG_MSM_KGSL
 static void load_gpu(struct drm_device *dev)
 {
 	static DEFINE_MUTEX(init_lock);
@@ -453,6 +454,11 @@ static void load_gpu(struct drm_device *dev)
 
 	mutex_unlock(&init_lock);
 }
+#else
+static void load_gpu(struct drm_device *dev)
+{
+}
+#endif
 
 static int msm_open(struct drm_device *dev, struct drm_file *file)
 {
@@ -1165,7 +1171,9 @@ static int __init msm_drm_register(void)
 	msm_dsi_register();
 	msm_edp_register();
 	hdmi_register();
+#ifndef CONFIG_MSM_KGSL
 	adreno_register();
+#endif
 	return platform_driver_register(&msm_platform_driver);
 }
 
@@ -1174,7 +1182,9 @@ static void __exit msm_drm_unregister(void)
 	DBG("fini");
 	platform_driver_unregister(&msm_platform_driver);
 	hdmi_unregister();
+#ifndef CONFIG_MSM_KGSL
 	adreno_unregister();
+#endif
 	msm_edp_unregister();
 	msm_dsi_unregister();
 }
-- 
2.8.2

